### ISSUE TITLE
Multiple child appenders in categoryFilter, logLevelFilter and multiprocess

### ISSUE THREAD

**Issue by florianreinhart** [log4js-node/log4js-node#518]
I run a multiprocess appender and want to filter the log messages on the master and send them to different appenders (stdout, file, smtp) based on the log level. Right now I have to define multiple master endpoints and filter in the workers.

It would be great if you could add multiple child appenders to the categoryFilter, logLevelFilter and multiprocess appenders.

A multiprocess config could like this:

```js
{
  appenders: {
    file: { type: 'file', filename: 'all-the-logs.log' },
    stdout: { type: 'stdout' },
    server: { type: 'multiprocess', mode: 'master', appenders: ['file', 'stdout'] }
  },
  categories: {
    default: { appenders: ['file', 'stdtout'], level: 'info' }
  }
}
```

Another more generic solution would be to create a new "multi appender", which just forwards log messages to multiple child appenders.

**Comment by nomiddlename** [comment]
If you're using this on a single machine, with multiple processes forked using `cluster` - then you'd probably be better off not using the multiprocess appender, and just letting log4js' cluster support handle it. 

If you're using the multiprocess appender to log over TCP from different processes/hosts, then I think for your use case (and probably most other users) maybe a re-write of the multiprocess appender would be a good idea. Adding multiple appenders to the multiprocess appender would mean you would lose a lot of the category and log-level based config that you would then have to re-create using logLevelFilters and categoryFilters. There could be a better way. Here's what I'm thinking of - on your master server, your log4js config would look like this:
```javascript
log4js.configure({
  appenders: {
    out: { type: 'stdout' },
    file: { type: 'file', filename: 'some-logs.log' }
  },
  categories: {
    default: { appenders: ['file', 'out'], level: 'info' }
  },
  listen: {
    host: '127.0.0.1',
    port: 1234
  }
});
```
You'd add that `listen` section to denote that you want to accept log messages over TCP - any messages that come in would get sent to the appenders using the same rules as local messages.

In your worker you would still need to define a tcp appender, something like this:
```javascript
log4js.configure({
  appenders: { net: { type: 'tcp', host: 'log.host', port: 1234 } },
  categories: { default: { appenders: ['net'], level: 'debug' } }
});
```
How does that sound? Would that solve your problem?

**Comment by florianreinhart** [comment]
Thanks a lot @nomiddlename!

That's a great idea and a way better solution 👍

**Comment by nomiddlename** [comment]
Cool - I'll let you know when it's ready to go, it'll take me a few days to code it up (I only have a few hours a week to work on this).

**Comment by nomiddlename** [comment]
This is turning into a major change, so I'm moving it to v3.

**Comment by nomiddlename** [comment]
Fixed in log4js@3.0.0 - the tcp client/server replaces the multiprocess appender

**Comment by florianreinhart** [comment]
@nomiddlename The functionality of tcp client/server is great and perfect for my use case. However, I ran into an issue when sending multiple log messages in quick succession, see #756.
